be/src/exec/vectorized/hdfs_scanner.h

```diff
@@ -186,7 +186,7 @@ class HdfsScanner {
     HdfsScanner() = default;
     virtual ~HdfsScanner() = default;
 
-    Status open(RuntimeState* runtime_state);
+    virtual Status open(RuntimeState* runtime_state);
```

Comment:
can you do customized stuff in `do_open`?

Reply:
`do_open` is supposed to be override.

Reply:
I override both of them for convenience because `HdfsScanner::open` method calls unexpected libhdfs `open`.

I can rethink this question and this part may need some tedious refactoring work.

Reply:
done, PTAL